docs/content/en/docs/implementing/tasks/_index.md

```diff
@@ -84,6 +97,101 @@ See the
 [KeptnTaskDefinition](../../yaml-crd-ref/taskdefinition.md)
 reference page for the synopsis and examples for each runner.
 
+## Executing sequential tasks
+
+All `KeptnTask` resources that are defined by
+`KeptnTaskDefinition` resources at the same level
+(either pre-deployment or post-deployment) execute in parallel.
+This is by design, because Keptn is not a pipeline engine.
+**Task sequences that are not part of the lifecycle workflow
+should not be handled by KLT**
+but should instead be handled by the pipeline engine tools being used
+such as Jenkins, Argo Workflows, Flux, and Tekton.
+
+If your lifecycle workflow includes
+a sequence of executables that need to be run in order,
+you can put them all in one `KeptnTaskDefinition` resource,
+which can execute a virtually unlimited number
+of programs, scripts, and functions,
+as long as they all need the same runner, such as Python.
```

Comment:
I would add a sentence to tell users they can also use their own language and provide a container :)
```suggestion
as long as they all need the same runner, such as Python.
Another option is to encode all your steps in the language of your choice and build a container that Keptn will execute.
```